src/modules/Builder/features/SaveAndPublish/SaveAndPublish.hooks.ts

```diff
@@ -198,19 +203,30 @@ export const useSaveAndPublishSetup = (hasPrompt: boolean) => {
     }
 
     setPublishProcessPopupOpened(false);
-    setIsPasswordPopupOpened(true);
+    const password = getPassword();
```

Comment:
could you pls create separate function for repeated code?

const password = getPassword();
    if (!password) {
      setIsPasswordPopupOpened(true);

      return;
    }
    await sendRequest(password);